tc_aws/loaders/s3_loader.py

```diff
@@ -32,6 +34,11 @@ def load(context, url, callback):
         def handle_data(file_key):
             if not file_key or 'Error' in file_key or 'Body' not in file_key:
                 logger.warn("ERROR retrieving image from S3 {0}: {1}".format(key, str(file_key)))
+                # We will now handle the failure and return a 404.
+                result = LoaderResult()
```

Comment:
Yes, I was considering that as an alternative option. I'll write the code to check for the amazon 404, and raise a ERROR_NOT_FOUND, otherwise raise an ERROR_UPSTREAM. Should be able to push it up tonight or tomorrow.